tarsier/tag_utils.ts

```diff
@@ -4,16 +4,22 @@ interface Window {
   // This means that subsequent calls to .evaluate will not have access to the functions defined in this file
   // since they will be in an inaccessible scope. To circumvent this, we attach the following methods to the
   // window which is always available globally when run in a browser environment.
-  tagifyWebpage: (tagLeafTexts?: boolean) => { [key: number]: string };
+  tagifyWebpage: (tagLeafTexts?: boolean) => TagMetadata[];
   removeTags: () => void;
   hideNonTagElements: () => void;
   revertVisibilities: () => void;
   fixNamespaces: (tagName: string) => string;
 }
 
 interface TagMetadata {
+  tarsierID: number;
+  elementName: string;
+  elementHTML: string;
```

Comment:
we actually only care about the opening tag, not the entire HTML (which would be pretty expensive to transfer)

Let's rename to `opening_tag_html`